awxkit/awxkit/ws.py

```diff
@@ -184,11 +184,16 @@ def _subscribe(self, **payload):
         payload['xrftoken'] = self.csrftoken
         self._send(json.dumps(payload))
 
-    def unsubscribe(self):
-        self._send(json.dumps(dict(groups={}, xrftoken=self.csrftoken)))
-        # it takes time for the unsubscribe event to be recieved and consumed and for
-        # messages to stop being put on the queue for daphne to send to us
-        time.sleep(5)
+    def unsubscribe(self, wait=True, timeout=10):
+        if wait:
+            # Other unnsubscribe events could have caused the edge to trigger.
+            # This way the _next_ event will trigger our waiting.
+            self._pending_unsubscribe.clear()
+            self._send(json.dumps(dict(groups={}, xrftoken=self.csrftoken)))
+            if not self._pending_unsubscribe.wait(timeout):
+                raise RuntimeError(f"Failed while waiting on unsubscribe reply because timeout of {timeout} seconds was reached.")
```

Comment:
This breaks awx CLI on RHEL7, please update to use `.format` as in other places in this file